discovery/consul/consul.go

```diff
@@ -47,6 +48,25 @@ func NewConsulConfig(config interface{}) (*Consul, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	maxRetry := 30
+	retry := 0
+	// we need to wait for Consul to start and self-elect
```

Comment:
This section appears to have been added to support https://github.com/joyent/containerpilot/issues/164 but I'd really like that behavior change to be a separate PR and design discussion.

I'm not comfortable adding a hard-coded retry here nor making it non-optional. It's entirely reasonable to have an application start up without Consul being ready, and to have it simply fail to make health checks until Consul is available (so long as ContainerPilot can handle this without crashing, which as far as I'm aware it can currently do just fine).